doc/neps/nep-0049.rst

```diff
@@ -0,0 +1,369 @@
+===================================
+NEP 49 — Data allocation strategies
+===================================
+
+:Author: Matti Picus
+:Status: Draft
+:Type: Standards Track
+:Created: 2021-04-18
+
+
+Abstract
+--------
+
+The ``numpy.ndarray`` requires additional memory allocations
+to hold ``numpy.ndarray.strides``, ``numpy.ndarray.shape`` and
+``numpy.ndarray.data`` attributes. These attributes are specially allocated
+after creating the python object in ``__new__`` method. The ``strides`` and
+``shape`` are stored in a piece of memory allocated internally.
```

Comment:
As per https://github.com/numpy/numpy/pull/18805#discussion_r616230941, can we rephrase/remove this paragraph please? I misread this too and thought the new allocator interface would change how `shape` and `strides` are allocated, which I couldn't care less.